src/Tool/Commands/BaseCommand.cs

```diff
@@ -9,25 +10,36 @@
 abstract class BaseCommand
 {
     readonly string[] maskNames;
-    readonly string outputPath;
+    readonly string reportName = "throughput-report";
     readonly bool isDevelopment;
 
     public BaseCommand(string[] maskNames)
     {
         this.maskNames = maskNames;
 
-        outputPath = Path.Combine(Environment.CurrentDirectory, "throughput-report.json");
-
         var envVars = Environment.GetEnvironmentVariables().Keys.OfType<string>().OrderBy(x => x).ToArray();
 
         if (!bool.TryParse(Environment.GetEnvironmentVariable("IS_DEVELOPMENT"), out isDevelopment))
         {
             isDevelopment = false;
         }
+    }
 
+    string GenerateReportTimeStamp(DateTime dateTime)
+    {
+        return $"{dateTime.Year}{dateTime.Month}-{dateTime.Day}-{dateTime.Hour}-{dateTime.Minute}-{dateTime.Second}";
     }
 
-    public async Task Run(CancellationToken cancellationToken = default)
+    string CreateReportOutputPath(string customerName)
+    {
+        var customerFileName = Regex.Replace(customerName, @"[^\w\d]+", "-").Trim('-').ToLower();
+        var outputPath = Path.Join(Environment.CurrentDirectory,
+            $"{customerFileName}-{reportName}-{GenerateReportTimeStamp(DateTime.Now)}");
```

Comment:
Still need this - for example `dateTime.Day` won't produce leading zero and the result would not be sortable.
```suggestion
            $"{customerFileName}-{reportName}-{DateTime.Now:yyyyMMdd-HHmmss}");
```